### ISSUE TITLE
Accessibility not working when legends enabled

### ISSUE THREAD

**Issue by ABINLESLYVA** [recharts/recharts#5100]
## Do you want to request a feature or report a bug?
Bug

## What is the current behavior?
I see that the `accessibilityLayer ` prop on a `LineChart ` which contains `<Legend />`  suppresses the keyboard navigation [arrow key] through data points

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via codesandbox or similar (template: https://codesandbox.io/p/sandbox/simple-line-chart-v25dl4).
You can find this behavior by adding `accessibilityLayer ` prop to the simple line chart from recharts examples. If I remove the Legend, it works.
https://recharts.org/en-US/examples 

## What is the expected behavior?
Accessibility / keyboard navigation is expected to work in all scenarios

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
Latest stable version - 2.12.7

**Comment by ckifer** [comment]
Hi, I cannot reproduce

https://codesandbox.io/p/sandbox/line-chart-width-xaxis-padding-forked-cvnpjp

Seems to be working fine here

**Comment by ABINLESLYVA** [comment]
My bad, I was pointing to an earlier version. Works well for me with the latest version. One question though, before closing this - does `ScatterChart `support accessibility layer ? If not, would switching to a `ComposedChart `make sense; Do you see any limitations here

**Comment by ckifer** [comment]
All of the Cartesian charts should support accessibility later. The polar ones, funnel, sunburst, treemap, and Sankey do not. Let me know if you have any issues
